### PR TITLE
Add 'consider using' message to overflowing_literals

### LINKED ISSUE #79744: `overflowing_literals` error is confusing because of type inference
While experimenting with the Rust By Example playground on the Inference section at https://doc.rust-lang.org/stable/rust-by-example/types/inference.html I came across what I believe is an error in the error produced by the compiler.

I do not expect this example to work but the error `literal out of range for 'i8'` is reported against line 4 `let e2 = 230;` which is a perfectly valid statement. The problem is lower down where two different types are pushed onto the same Vector but possibly due to the way the inference engine works this is being incorrectly reported against line 4. If I comment out the line `vec.push(e2);`, the example works fine, apart from a `warning: unused variable: 'e2'`.

### Code

```Rust
fn main() {
    // Because of the annotation, the compiler knows that `elem` has type i8.
    let elem = 6i8;
    let e2 = 230;

    // Create an empty vector (a growable array).
    let mut vec = Vec::new();
    // At this point the compiler doesn't know the exact type of `vec`, it
    // just knows that it's a vector of something (`Vec<_>`).

    // Insert `elem` in the vector.
    vec.push(e2);
    vec.push(elem);
    // Aha! Now the compiler knows that `vec` is a vector of `u8`s (`Vec<u8>`)
    // TODO ^ Try commenting out the `vec.push(elem)` line

    println!("{:?}", vec);
}

```

### Meta

`rustc --version --verbose`: I am unable to determine the exact compiler version in the Rust By Example playground. I have run the same example on my Fedora 32 Workstation machine and with the same result. The rustc version details for this machine are as follows
```
rustc 1.48.0
binary: rustc
commit-hash: unknown
commit-date: unknown
host: x86_64-unknown-linux-gnu
release: 1.48.0
LLVM version: 10.0

```

### Error output

```
   Compiling playground v0.0.1 (/playground)
error: literal out of range for `i8`
 --> src/main.rs:4:14
  |
4 |     let e2 = 230;
  |              ^^^
  |
  = note: `#[deny(overflowing_literals)]` on by default
  = note: the literal `230` does not fit into the type `i8` whose range is `-128..=127`

error: aborting due to previous error

error: could not compile `playground`

To learn more, run the command again with --verbose.
```

<details><summary><strong>Backtrace</strong></summary>
Sorry I don't know exactly how to provide this.
<p>

```
<backtrace>
```

</p>
</details>

### PR DESCRIPTION
**Pull request by camelid**
Fixes #79744.

Ironically, the `overflowing_literals` handler for binary or hex already
had this message! You would think it would be the other way around :)

cc @scottmcm

### CODE REVIEW

**Comment by camelid** [comment]
Note that this doesn't produce a suggestion yet; that's next.

File: compiler/rustc_lint/src/types.rs [hunk 1]
```diff
@@ -331,18 +331,23 @@ fn lint_int_literal<'tcx>(
...
+                    .span_to_snippet(lit.span)
+                    .expect("must get snippet from literal"),
+                t.name_str(),
+                min,
+                max,
+            ));
+            if let Some(sugg_ty) =
+                get_type_suggestion(&cx.typeck_results().node_type(e.hir_id), v, negative)
+            {
+                err.help(&format!("consider using `{}` instead", sugg_ty));
```
Comment (lcnr) [inline compiler/rustc_lint/src/types.rs:348]: "using `i8`" sounds weird to me, maybe something like
```suggestion
                err.help(&format!("consider changing its type to `{}`", sugg_ty));
```

**Review by lcnr — APPROVED** [review]
Is there something I can do here rn? I don't think that a suggestion is strictly necessary and would be fine with landing it as is if you are also fine with doing so

File: compiler/rustc_lint/src/types.rs (hunk 1 above)
Comment (camelid) [reply compiler/rustc_lint/src/types.rs:348]: I just copied-and-pasted this from the binary/hexadecimal error. If we change this one, we should probably change the other one too.

Comment (lcnr) [reply compiler/rustc_lint/src/types.rs:348]: hmm, yeah I would prefer us to change this as well? Do you have a preference here?

Comment (camelid) [reply compiler/rustc_lint/src/types.rs:348]: What about

```suggestion
                err.help(&format!("consider using the `{}` type instead", sugg_ty));
```

or something like that?

Comment (lcnr) [reply compiler/rustc_lint/src/types.rs:348]:
```suggestion
                err.help(&format!("consider using the type `{}` instead", sugg_ty));
```

Comment (camelid) [reply compiler/rustc_lint/src/types.rs:348]: There are things like:

```rust
            ty::Char => FfiUnsafe {
                ty,
                reason: "the `char` type has no C equivalent".into(),
                help: Some("consider using `u32` or `libc::wchar_t` instead".into()),
            },
```

and

```rust
            ty::Str => FfiUnsafe {
                ty,
                reason: "string slices have no C equivalent".into(),
                help: Some("consider using `*const u8` and a length instead".into()),
            },
```

but those are for a different error so leaving them be.

**Pull request merged**